### PR TITLE
parts: add early base parsing (CRAFT-766)

### PR DESCRIPTION
**Pull request by cmatsuoka**
Load the project YAML file to verify the base entry when a lifecycle
command is used. If it's not core22, fall back to the legacy handlers.
    
Signed-off-by: Claudio Matsuoka <claudio.matsuoka@canonical.com>

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: snapcraft/parts/lifecycle.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    emit.trace(f"{step_name} parts, arguments: {parsed_args}")
+    yaml_data = {}
+    for project_file in _PROJECT_FILES:
+        if project_file.is_file():
+            yaml_data = _load_yaml(project_file)
+            break
+    else:
+        raise errors.SnapcraftError(
+            "Could not find snap/snapcraft.yaml. Are you sure you are in the "
+            "right directory?\nTo start a new project, use `snapcraft init`"
```
Comment (facundobatista) [inline snapcraft/parts/lifecycle.py:56]: Don't compose multiline error messages. 

Instead, pass the "resolution" as intended:
```
raise errors.SnapcraftError(
            "Could not find snap/snapcraft.yaml. Are you sure you are in the right directory?"
            resolution="To start a new project, use `snapcraft init`")
```

**Review by facundobatista — DISMISSED** [review]
Nice! One comment inline, though. Thanks!

File: snapcraft/parts/lifecycle.py (hunk 1 above)
Comment (cmatsuoka) [reply snapcraft/parts/lifecycle.py:56]: Thanks, will fix that

**Review by facundobatista — APPROVED** [review]
+1

**Pull request merged**